caas/kubernetes/provider/resources/applier.go

```diff
@@ -52,29 +56,73 @@ func (op *operation) process(ctx context.Context, rollback Applier) error {
 	} else if err != nil {
 		return errors.Annotatef(err, "checking if resource %q exists or not", existingRes.ID())
 	}
-	if found {
-		ver := op.resource.GetObjectMeta().GetResourceVersion()
-		if ver != "" && ver != existingRes.GetObjectMeta().GetResourceVersion() {
-			id := op.resource.ID()
-			return errors.Annotatef(errConflict, "%s %s", id.Type, id.Name)
-		}
-	}
+
 	switch op.opType {
 	case opApply:
-		err = op.resource.Apply(ctx)
-		if found {
-			// apply the previously existing resource.
-			rollback.Apply(existingRes)
-		} else {
-			// delete the new resource just created.
-			rollback.Delete(op.resource)
+		err = retry.Call(retry.CallArgs{
+			Func: func() error {
+				err := op.resource.Apply(ctx)
+				if errors.Is(err, errConflict) {
+					_ = op.resource.Get(ctx) // refresh resource version
+					return err
+				}
+				// this might happen if it were deleted by another apiserver.
+				if k8serrors.IsNotFound(err) {
+					return err
+				}
+				return errors.Annotatef(err, "applying resource %q", op.resource.ID().Name)
+			},
+			IsFatalError: func(err error) bool {
+				return !k8serrors.IsConflict(err) && !errors.Is(err, errConflict)
```

Comment:
We only ever would get `errConflict` right? So we don't need `k8serrors.IsConflict(err)`

Reply:
Add a comment for apply for returning err